src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,137 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> first;
+    private Node<T> last;
+    private int size;
+
+    private static class Node<T> {
+        private T value;
+        private Node<T> next;
+        private Node<T> prev;
+
+        Node(Node<T> prev, T value, Node<T> next) {
+            this.value = value;
+            this.next = next;
+            this.prev = prev;
+        }
+    }
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(last, value, null);
+        if (size == 0) {
+            first = newNode;
+        } else {
+            last.next = newNode;
+        }
+        last = newNode;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new ArrayIndexOutOfBoundsException();
+        } else if (index == size) {
+            add(value);
+        } else if (index == 0) {
+            Node<T> newNode = new Node<>(null, value, first);
+            newNode.prev = newNode;
+            first = newNode;
+            size++;
+        } else {
+            Node<T> searchIndex = searchIndex(index);
+            Node<T> newNode = new Node<>(searchIndex.prev, value, searchIndex);
+            searchIndex.prev.next = newNode;
+            searchIndex.prev = newNode;
+            size++;
+        }
```

Comment:
refactor code accordingly
```suggestion
        }
        if (index == size) {
            lintTail(value);
            return;
        }
        if (index == 0) {
            linkHead(value);
            return;
        }
            Node<T> searchIndex = searchIndex(index);
            Node<T> newNode = new Node<>(searchIndex.prev, value, searchIndex);
            searchIndex.prev.next = newNode;
            searchIndex.prev = newNode;
            size++;
```